src/Compilers/CSharp/Portable/Symbols/Retargeting/RetargetingMethodSymbol.cs

```diff
@@ -219,6 +222,43 @@ public override ImmutableArray<CSharpAttributeData> GetReturnTypeAttributes()
             return this.RetargetingTranslator.GetRetargetedAttributes(_underlyingMethod.GetReturnTypeAttributes(), ref _lazyReturnTypeCustomAttributes);
         }
 
+#nullable enable
+        internal override UnmanagedCallersOnlyAttributeData? UnmanagedCallersOnlyAttributeData
+        {
+            get
+            {
+                if (_lazyUnmanagedAttributeData == UnmanagedCallersOnlyAttributeData.Uninitialized)
+                {
+                    var underlyingData = _underlyingMethod.UnmanagedCallersOnlyAttributeData;
+                    if (underlyingData == UnmanagedCallersOnlyAttributeData.Uninitialized)
+                    {
+                        // Underlying hasn't been found yet either, just return it. We'll check again the next
+                        // time this is called
+                        return underlyingData;
+                    }
+
+                    if (underlyingData == null || (underlyingData.CallingConventionTypes?.IsEmpty != false))
+                    {
+                        _ = Interlocked.CompareExchange(ref _lazyUnmanagedAttributeData, underlyingData, UnmanagedCallersOnlyAttributeData.Uninitialized);
+                        return _lazyUnmanagedAttributeData;
+                    }
+
+                    var builder = PooledHashSet<INamedTypeSymbolInternal>.GetInstance();
+                    foreach (var identifier in underlyingData.CallingConventionTypes)
+                    {
+                        builder.Add((INamedTypeSymbolInternal)RetargetingTranslator.Retarget((NamedTypeSymbol)identifier));
+                    }
+
+                    var newData = builder.Count == 0 ? ImmutableHashSet<INamedTypeSymbolInternal>.Empty : builder.ToImmutableHashSet();
```

Comment:
>builder.Count == 0 [](start = 34, length = 18)

Is this ever `true`?

Reply:
Yes, if there were no conventions specified.

---
In reply to: [482201192](https://github.com/dotnet/roslyn/pull/47311#discussion_r482201192) [](ancestors = 482201192)

Reply:
It looks like the case where `CallingConventions.IsEmpty` is handled earlier.

---
In reply to: [482280916](https://github.com/dotnet/roslyn/pull/47311#discussion_r482280916) [](ancestors = 482280916,482201192)